contribs/gmf/src/directives/displayquerygrid.js

```diff
@@ -0,0 +1,788 @@
+     * @param {angular.Attributes} attrs Attributes.
+     * @return {string} Template.
+     */
+    function(element, attrs) {
+      var templateUrl = attrs['gmfDisplayquerygridTemplateurl'];
+      return templateUrl !== undefined ? templateUrl :
+          gmf.baseTemplateUrl + '/displayquerygrid.html';
+    });
+
+
+/**
+ * Provides a directive to display results of the {@link ngeo.queryResult} in a
+ * grid and shows related features on the map using
+ * the {@link ngeo.FeatureOverlayMgr}.
+ *
+ * You can override the default directive's template by setting the
+ * value `gmfDisplayquerygridTemplateUrl`.
+ *
+ * Features displayed on the map use a default style but you can override these
+ * styles by passing ol.style.Style objects as attributes of this directive.
+ *
+ * Example:
+ *
+ *      <gmf-displayquerygrid
+ *        gmf-displayquerygrid-map="ctrl.map"
+ *        gmf-displayquerygrid-featuresstyle="ctrl.styleForAllFeatures"
+ *        gmf-displayquerygrid-selectedfeaturestyle="ctrl.styleForTheCurrentFeature">
+ *      </gmf-displayquerygrid>
+ *
+ * @htmlAttribute {boolean} gmf-displayquerygrid-active The active state of the component.
+ * @htmlAttribute {ol.style.Style} gmf-displayquerygrid-featuresstyle A style
+ *     object for all features from the result of the query.
+ * @htmlAttribute {ol.style.Style} gmf-displayquerygrid-selectedfeaturestyle A style
+ *     object for the currently selected features.
+ * @htmlAttribute {ol.Map} gmf-displayquerygrid-map The map.
+ * @htmlAttribute {boolean?} gmf-displayquerygrid-removeemptycolumns Optional. Should
+ *     empty columns be hidden? Default: `false`.
+ * @htmlAttribute {number?} gmf-displayquerygrid-maxrecenterzoom Optional. Maximum
+ *     zoom-level to use when zooming to selected features.
+ * @htmlAttribute {gmfx.GridMergeTabs?} gmf-displayquerygrid-gridmergetabas Optional.
+ *     Configuration to merge grids with the same attributes into a single grid.
+ * @param {string} gmfDisplayquerygridTemplateUrl URL to a template.
+ * @return {angular.Directive} Directive Definition Object.
+ * @ngInject
+ * @ngdoc directive
+ * @ngname gmfDisplayquerygrid
+ */
+gmf.displayquerygridDirective = function(
+    gmfDisplayquerygridTemplateUrl) {
+  return {
+    bindToController: true,
+    controller: 'GmfDisplayquerygridController',
+    controllerAs: 'ctrl',
+    templateUrl: gmfDisplayquerygridTemplateUrl,
+    replace: true,
+    restrict: 'E',
+    scope: {
+      'active': '=gmfDisplayquerygridActive',
+      'featuresStyleFn': '&gmfDisplayquerygridFeaturesstyle',
+      'selectedFeatureStyleFn': '&gmfDisplayquerygridSourceselectedfeaturestyle',
+      'getMapFn': '&gmfDisplayquerygridMap',
+      'removeEmptyColumnsFn': '&?gmfDisplayquerygridRemoveemptycolumns',
+      'maxResultsFn': '&?gmfDisplayquerygridMaxresults',
+      'maxRecenterZoomFn': '&?gmfDisplayquerygridMaxrecenterzoom',
+      'mergeTabsFn': '&gmfDisplayquerygridMergetabs'
```

Comment:
That should not be `&?gmfDisplayquerygridMergetabs` ? (bound with `&?` instead of `&`) ?